service/worker/archiver/client.go

```diff
@@ -56,6 +57,11 @@ type (
 		ArchiveRequest       *ArchiveRequest
 		CallerService        string
 		AttemptArchiveInline bool
+		// SkipRetrySignal will skip sending a signal to the archival workflow to retry archival if the inline attempt
+		// fails. Note that this does nothing if AttemptArchiveInline is false. We will set this to true for durable
+		// archival because the archival queue handles retries, and then we can  remove this condition when the old
+		// behavior is deleted.
+		SkipRetrySignal bool
```

Comment:
Can the send signal logic remove after durable archival? Is this flag needed?

Reply:
This flag is needed, but we can remove it after durable archival is fully rolled out.

Reply:
Once we have durable archival, the code should be moved to history service. and archival client probably should not exist at all.

so I guess, we can either 
- add the flag, use the client for now in archival queue, then later refactor. 
- Or just copy/paste the inline archival code and use it in archival task executor. Then later we can simply delete the entire archival client. 
Either way we'll need to do some refactoring. so I don't have strong opinion on which one we should pick.

Reply:
Yeah. I would vote for the later. With refactoring, this seems a throwing work.